sdks/python/apache_beam/transforms/timeutil.py

```diff
@@ -64,19 +64,19 @@ class TimestampCombinerImpl(with_metaclass(ABCMeta, object)):
 
   @abstractmethod
   def assign_output_time(self, window, input_timestamp):
-    pass
+    raise NotImplementedError
 
   @abstractmethod
   def combine(self, output_timestamp, other_output_timestamp):
-    pass
+    raise NotImplementedError
 
   def combine_all(self, merging_timestamps):
     """Apply combine to list of timestamps."""
     combined_output_time = None
     for output_time in merging_timestamps:
       if combined_output_time is None:
         combined_output_time = output_time
-      else:
+      elif output_time is not None:
         combined_output_time = self.combine(
             combined_output_time, output_time)
```

Comment:
what would be the "else" after this change, What would it imply when this happens?   Shall we at least log something when fall into that situation?

Reply:
Currently the code can handle `None`s up until the first non-None value, and not after that. This makes it more consistent by allowing any values to be `None`. The else is to go onto the next value.